### PR TITLE
Add live metrics api

### PR DESCRIPTION
**Pull request by wiktork**
Addresses #68 

Sample output:
``` json
{"timestamp":"2021-08-20T13:59:12.831569-07:00","provider":"System.Runtime","name":"cpu-usage","displayName":"CPU Usage","unit":"%","counterType":"Metric","value":0}
{"timestamp":"2021-08-20T13:59:12.831684-07:00","provider":"System.Runtime","name":"working-set","displayName":"Working Set","unit":"MB","counterType":"Metric","value":66}
{"timestamp":"2021-08-20T13:59:12.8321364-07:00","provider":"System.Runtime","name":"gc-heap-size","displayName":"GC Heap Size","unit":"MB","counterType":"Metric","value":4}

```

TODO: 

- [ ] Documentation
- [x] There is an occasionally issue in the tests where the data comes back partial. This only reproes in test environments.
- [ ] Figure out other interesting export formats

Sample output from System.Diagnostics.Metrics counters from dotnet-counters tool for comparison:

```json
 "Events": [
        {
            "timestamp": "2021-08-31 09:47:54Z",
            "provider": "name1",
            "name": "name2 (u1 / 1 sec)",
            "tags": "id=00-91a528e9074c0a8d796833a63d9bb4ea-eaf508e5a4460c92-00",
            "counterType": "Rate",
            "value": 1.2323
        },
        {
            "timestamp": "2021-08-31 09:47:54Z",
            "provider": "name1",
            "name": "name3 (h)",
            "tags": "Percentile=50",
            "counterType": "Metric",
            "value": 94
        },
        {
            "timestamp": "2021-08-31 09:47:54Z",
            "provider": "name1",
            "name": "name3 (h)",
            "tags": "Percentile=95",
            "counterType": "Metric",
            "value": 94
        },
 ]
```

### CODE REVIEW

**Comment by kelltrick** [comment]
Please add entry to [/documentation/releaseNotes.md](https://github.com/dotnet/dotnet-monitor/blob/main/documentation/releaseNotes/releaseNotes.md)

File: src/Microsoft.Diagnostics.Monitoring.WebApi/Controllers/DiagController.Metrics.cs [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+        /// <summary>
+        /// Capture live metrics for a process.
+        /// </summary>
+        /// <param name="pid">Process ID used to identify the target process.</param>
+        /// <param name="uid">The Runtime instance cookie used to identify the target process.</param>
+        /// <param name="name">Process name used to identify the target process.</param>
+        /// <param name="durationSeconds">The duration of the metrics session (in seconds).</param>
+        /// <param name="metricsIntervalSeconds">The reporting interval (in seconds) for event counters.</param>
+        /// <param name="egressProvider">The egress provider to which the metrics are saved.</param>
+        [HttpGet("livemetrics", Name = nameof(LiveMetrics))]
```
[the remark this answers is not in the document]
Comment (jander-msft) [reply src/Microsoft.Diagnostics.Monitoring.WebApi/Controllers/DiagController.Metrics.cs:33]: Per offline discussion, suggested route name is `/collectmetrics`.

File: src/Tests/Microsoft.Diagnostics.Monitoring.Tool.FunctionalTests/LiveMetricsTests.cs [hunk 2]
```diff
@@ -30,27 +30,28 @@
 namespace Microsoft.Diagnostics.Monitoring.Tool.FunctionalTests
 {
     [Collection(DefaultCollectionFixture.Name)]
-    public class LiveMetricsTests
+    public class CollectMetricsTests
+
```
Comment (jander-msft) [inline src/Tests/Microsoft.Diagnostics.Monitoring.Tool.FunctionalTests/LiveMetricsTests.cs:34]: nit: extra newline?

**Review by jander-msft — APPROVED** [review]

**Pull request merged**